File: p009.py
# 
# Solution to Project Euler problem 9
# by Project Nayuki
# 
# http://www.nayuki.io/page/project-euler-solutions
# https://github.com/nayuki/Project-Euler-solutions
# 


def compute():
	LIMIT = 1000
	for a in range(1, LIMIT + 1):
		for b in range(a, LIMIT + 1):
			c = LIMIT - a - b
			if c >= b:
				# Now a <= b <= c
				if a**2 + b**2 == c**2:
					return str(a * b * c)


if __name__ == "__main__":
	print(compute())

File: haskell/p009.hs
{- 
 - Solution to Project Euler problem 9
 - Copyright (c) Project Nayuki. All rights reserved.
 - 
 - https://www.nayuki.io/page/project-euler-solutions
 - https://github.com/nayuki/Project-Euler-solutions
 -}


-- Computers are fast, so we can implement a brute-force search to directly solve the problem.
perim = 1000
main = putStrLn (show ans)
ans = head [a * b * (perim - a - b) | a <- [1..perim], b <- [a+1..perim], isIntegerRightTriangle a b]
isIntegerRightTriangle a b = a < b && b < c
	&& a * a + b * b == c * c
	where c = perim - a - b
